src/server/internalServer.cpp

```diff
@@ -141,7 +149,9 @@ InternalServer::InternalServer(Library* library,
   m_indexTemplateString(indexTemplateString.empty() ? RESOURCE::templates::index_html : indexTemplateString),
   mp_daemon(nullptr),
   mp_library(library),
-  mp_nameMapper(nameMapper ? nameMapper : &defaultNameMapper)
+  mp_nameMapper(nameMapper ? nameMapper : &defaultNameMapper),
+  searcherCache(std::max(getCacheLength("SEARCHER_CACHE_SIZE", mp_library->getBookCount(true, true)*0.1), 1U)),
+  searchCache(std::max(getCacheLength("SEARCH_CACHE_SIZE", SEARCH_CACHE_SIZE), 1U))
```

Comment:
It should be valid for the user to set the env variable `SEARCH_CACHE_SIZE` to 0 (and so deactivate the cache).
So the searchCache should not have any `max`. And the `max`should be applied to the `getBookCount` only :
```cpp
searcherCache(getCacheLength("SEARCHER_CACHE_SIZE", std::max(mp_library->getBookCount(true, true)*0.1), 1U)),
searchCache(getCacheLength("SEARCH_CACHE_SIZE", SEARCH_CACHE_SIZE))
```